token/jwt/token_test.go

```diff
@@ -0,0 +1,47 @@
+package jwt
```

Comment:
yes, I took most of `parse_test.go` test cases, I prune those we do not need or support e.g. `json.Number` and added specific code to prevent regresses like returning always a toke instance and returning a specific validation error when a Key was not provided, the rest worked perfect. I also appended the test cases that you mentioned.